Optimize hibernate use in AsyncJobStatusCuratorTest

barnabycourt opened the pull request:
General updates to the AsyncJobStatusCuratorTest unit tests. Locally, this reduced runtime from over 20 minutes to ~30 seconds.  
* Remove extraneous flushes
* Re-organize the createJobsForQueryTests() in order to reduce database load / time to execute.

Ceiu commented:
I took the idea and did a bit of optimization and cleanup on it [1]. I'd be curious to hear your take on it and see if it passes with the same level of efficiency.

[1] https://gist.github.com/Ceiu/781b551e8c46dd59fd18dfdb8900dc73

barnabycourt commented:
@Ceiu I updated based on some of the suggestions in the gist. I made some changes to the suggestions for the createJob() method in order to drop an unused parameter & make sure that the block on flush was passed through to the generic create method. 

I removed the block size from the in clause as HSQLDB does not have an in clause limit & this call is only used in testing. I tried setting a 1k limit for the in clause in HSQLDB but that seemed to cause general slowdowns in other parts of the suite.
